dawn/src/dawn/AST/ASTExpr.cpp

```diff
@@ -445,16 +446,26 @@ bool LiteralAccessExpr::equals(const Expr* other) const {
          (data_ ? data_->equals(otherPtr->data_.get()) : !otherPtr->data_);
 }
 
-ReductionOverNeighborExpr::ReductionOverNeighborExpr(std::string const& op,
-                                                     std::shared_ptr<Expr> const& rhs,
-                                                     std::shared_ptr<Expr> const& init,
-                                                     ast::Expr::LocationType rhs_location,
-                                                     SourceLocation loc)
-    : Expr(Kind::ReductionOverNeighborExpr, loc), op_(op),
+//===------------------------------------------------------------------------------------------===//
+//     ReductionOverNeighborExpr
+//===------------------------------------------------------------------------------------------===//
+
+ReductionOverNeighborExpr::ReductionOverNeighborExpr(
+    std::string const& op, std::shared_ptr<Expr> const& rhs, std::shared_ptr<Expr> const& init,
+    ast::Expr::LocationType lhs_location, ast::Expr::LocationType rhs_location, SourceLocation loc)
+    : Expr(Kind::ReductionOverNeighborExpr, loc), op_(op), lhs_location_(lhs_location),
       rhs_location_(rhs_location), operands_{rhs, init} {}
 
+ReductionOverNeighborExpr::ReductionOverNeighborExpr(
+    std::string const& op, std::shared_ptr<Expr> const& rhs, std::shared_ptr<Expr> const& init,
+    std::vector<std::shared_ptr<sir::Value>> weights, ast::Expr::LocationType lhs_location,
```

Comment:
this was because `Value` is non copyable (since its purpose was to represent globals). This address the situation, `Value` was made copyable and a derived class `Global` was introduced that is non copyable. The `Global` class is now used to represent globals.